File: mailer/event/consume.py
import json
import logging
import os

from rabbit_helper.rabbit import Rabbit
from send_mail import send_mail


async def consume(routing_key):
    rabbitmq_url = f"amqp://{os.getenv('RABBITMQ_USERNAME')}:{os.getenv('RABBITMQ_PASSWORD')}@{os.getenv('RABBITMQ_HOST')}:{os.getenv('RABBITMQ_PORT')}/"
    rabbit = Rabbit(rabbitmq_url)

    await rabbit.consume(routing_key, callback)


def callback(ch, method, properties, body):
    data = json.loads(body)
    send_mail(data)
    logging.info(f"Received message: {data}")

File: logger/event/consume.go
package event

import (
	"fmt"
	Rabbit "github.com/arshia-rgh/rabbit-helper-go/rabbit_helper"
	"os"
)

func Consume(routingKey string, callback func(data any)) {
	rabbitMQURL := fmt.Sprintf("amqp://%v:%v@%v:%v/",
		os.Getenv("RABBITMQ_USERNAME"),
		os.Getenv("RABBITMQ_PASSWORD"),
		os.Getenv("RABBITMQ_HOST"),
		os.Getenv("RABBITMQ_PORT"),
	)

	rabbit := Rabbit.New(rabbitMQURL)
	defer rabbit.Close()

	go func() {
		err := rabbit.Consume(routingKey, callback)
		if err != nil {
			return
		}
	}()
}
